src/NServiceBus.RabbitMQ.Tests/MessageConverterTests.cs

```diff
@@ -147,36 +146,71 @@ public void TestCanHandleStringHeader()
                     MessageId = "Blah",
                     Headers = new Dictionary<string, object>
                     {
-                        {"Foo", "ni"}
+                        {"short", 42},
+                        {"int", 42},
+                        {"long", 42},
+                        {"decimal", 42m},
+                        {"sbyte", (sbyte)42},
+                        {"double", 42d},
+                        {"single", 42f},
+                        {"bool", true }
                     }
                 }
             };
 
             var headers = converter.RetrieveHeaders(message);
 
             Assert.NotNull(headers);
-            Assert.AreEqual("ni", headers["Foo"]);
+            Assert.AreEqual("42", headers["short"]);
+            Assert.AreEqual("42", headers["int"]);
+            Assert.AreEqual("42", headers["long"]);
+            Assert.AreEqual("42", headers["decimal"]);
+            Assert.AreEqual("42", headers["sbyte"]);
+            Assert.AreEqual("42", headers["double"]);
+            Assert.AreEqual("42", headers["single"]);
+            Assert.AreEqual("True", headers["bool"]);
         }
```

Comment:
The reason I wrote the test the way I did is that all of those values exercise the exact same code path in `ValueToString` : the `value?.ToString()` call at the end. I didn't really see a lot of value in having a bunch of different tests.

The test is actually making sure we can handle all of the possible AMQP field types of a field table, so perhaps `TestCanHandleHeadersWithAllAmqpFieldValues.`

>The short and long cases are not properly typed, they should be (short)42 and 42L.

Good call on that, missed those.